lib/mongoid/errors/unmapped_bson_type.rb

```diff
@@ -0,0 +1,22 @@
+# encoding: utf-8
+module Mongoid
+  module Errors
+
+    # This error is raised when attempting to demongoize a value from the database
+    # that doesn't map to the field type.
+    class UnmappedBSONType < MongoidError
```

Comment:
Are we keeping this error around? See Emily's comment [here](https://github.com/mongodb/mongoid/pull/4363#discussion_r93001937)

Reply:
From what I am understanding, I don't think so. If user specifies BigDecimal as field type + Decimal128 serialization (which would have to be explicitly enabled at least in the first Mongoid release incorporating the integration) they are accepting the common functionality of the two types, including the fact that trailing zeroes will be lost in a roundtrip.

Reply:
done